src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let preValue = initialValue;
+
+    if (arguments.length < 2) {
```

Comment:
Instead of checking 'arguments.length', consider checking if 'initialValue' is undefined.

Reply:
if i change this statmet my code will return error when undefined was passed as initialValue